### PR TITLE
feat(sdk): enable loading both JSON and YAML pipelines IR

### LINKED ISSUE #1088: Support YAML Pipelines IR serialization format
**Description**
We want to support the new preferred Pipelines IR serialization format: YAML.

**Reason**
Without this change, [kubeflow/pipelines/pull/7431](https://github.com/kubeflow/pipelines/pull/7431) will be a breaking change for `aiplatform`.

### PR DESCRIPTION
**Pull request by connor-mccarthy**
Fixes #1088

**Main changes**
This PR switches `load_json` to `load_yaml`. This change is backward-compatible, as YAML is a superset of JSON.

**Other changes**
* Added tests for `load_yaml` to demonstrate that is reads both JSON and YAML.
* Added `pyyaml` as a library dependency.
* Updated copyright where applicable.
* Formatted `tests/unit/aiplatform/test_utils.py` with `black`.

**Manually tested the four primary codepaths**
1. Read JSON locally.
2. Read YAML locally.
3. Read JSON from Google Cloud Storage.
4. Read YAML from Google Cloud Storage.

---

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

cc: @chensun @ji-yaqi

### CODE REVIEW

File: google/cloud/aiplatform/utils/yaml_utils.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 
-# Copyright 2020 Google LLC
+# Copyright 2022 Google LLC
```
Comment (chensun) [inline google/cloud/aiplatform/utils/yaml_utils.py:3]: nit: preserve the initial year, or leave it untouched.
https://opensource.google/documentation/reference/copyright#the_year

**Review by chensun — COMMENTED** [review]
Consider adding some YAML snippets here along side the existing JSON one?

https://github.com/googleapis/python-aiplatform/blob/52273c2108c9bb24eadab214036f2ef93b847321/tests/unit/aiplatform/test_pipeline_jobs.py#L67-L104

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 
-# Copyright 2021 Google LLC
+# Copyright 2021-2022 Google LLC
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/pipeline_jobs.py:3]: @parthea Is this the correct formatting or should it just be `2022`. It seems like GAPIC changes to the latest year.

Comment (sasha-gitg) [reply google/cloud/aiplatform/pipeline_jobs.py:3]: Take a look at @chensun comment below

Comment (parthea) [reply google/cloud/aiplatform/pipeline_jobs.py:3]: For autogenerated files it should be `# Copyright 2022 Google LLC`. For manually updated files we should leave the original copyright year untouched as mentioned by @chensun.

File: tests/unit/aiplatform/test_utils.py [hunk 3]
```diff
@@ -515,3 +513,33 @@ def test_tensorboard_get_experiments_compare_url_fail_diff_region(self):
...
+    json_file_path = os.path.join(tmp_path, "test.json")
+    with open(json_file_path, "w") as f:
+        json.dump(data, f)
+    yield json_file_path
+
+
+class TestYamlUtils:
+    def test_load_yaml_from_local_file__with_json(self, yaml_file):
+        data = yaml_utils.load_yaml(yaml_file)
+        assert data["key"] == "val"
```
Comment (sasha-gitg) [inline tests/unit/aiplatform/test_utils.py:539]: I think pytest supports dict equality with `assert d1 == d2`

**Review by sasha-gitg — CHANGES_REQUESTED** [review]

**Review by sasha-gitg — APPROVED** [review]

File: tests/unit/aiplatform/test_pipeline_jobs.py [hunk 4]
```diff
@@ -226,10 +279,11 @@ def mock_pipeline_service_list():
 
 
 @pytest.fixture
-def mock_load_json(job_spec_json):
-    with patch.object(storage.Blob, "download_as_bytes") as mock_load_json:
-        mock_load_json.return_value = json.dumps(job_spec_json).encode()
-        yield mock_load_json
+def mock_load_yaml_and_json(job_spec):
+    with patch.object(storage.Blob, "download_as_bytes") as mock_load_yaml_and_json:
+        job_spec_str = json.dumps(job_spec) if isinstance(job_spec, dict) else job_spec
```
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply tests/unit/aiplatform/test_pipeline_jobs.py:284]: Definitely. That originates from the inconsistent starting point of the two representations in the tests: the JSON starts as a dict and the YAML starts as a string. In real usage, they would likely both be read in as strings from a file.

One option to make it more consistent is to put the json/dicts into strings as well, either literally with `"""` or programmatically (my preference) by wrapping in `json.dumps` at definition. If not one of those two options, I'll add a comment. WDYT?

[the remark this answers is not in the document]
Comment (sasha-gitg) [reply tests/unit/aiplatform/test_pipeline_jobs.py:284]: `json.dumps` SGTM. Thanks!

**Comment by connor-mccarthy** [comment]
@sasha-gitg, can you please re-add the automerge label if all looks good?

**Pull request merged**